.github/workflows/rust.yml

```diff
@@ -0,0 +1,80 @@
+name: Rust
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+
+  clippy:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-clippy
+    - name: Run clippy
+      run: |
+        cargo clippy --all-features --no-deps -p near-cache -p near-performance-metrics -p near-performance-metrics-macros -p near-rate-limiter -p near-stable-hasher -- -D clippy::all
+
+  cargo-deny:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      id: deny-cache
```

Comment:
cargo deny has no build artifacts, there's no need to cache. If anything, we only need to keep track of changes to the lockfiles.

Reply:
@miraclx The caching is for the installation of `cargo-deny`. It's not preinstalled, and it takes 6m to install.